flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCTableSourceSinkFactory.java

```diff
@@ -136,6 +146,35 @@
 		return builder.build();
 	}
 
+	/**
+	 * The original table schema may contain generated columns which shouldn't be produced/consumed
+	 * by TableSource/TableSink. And the original TIMESTAMP/DATE/TIME types uses LocalDateTime/LocalDate/LocalTime
+	 * as the conversion classes, however, JDBC connector uses Timestamp/Date/Time classes. So that
+	 * we bridge them to the expected conversion classes.
+	 */
+	private TableSchema getPhysicalTableSchema(DescriptorProperties descriptorProperties) {
```

Comment:
Should there be a util method in `DescriptorProperties` or somewhere to re-use?

Reply:
I don't think it is a general util to be in `DescriptorProperties`. I think it is more connector-specific behavior.

Reply:
Every connector should only want physical schema, right?

Reply:
You are right. But I would like to do it in another JIRA (FLINK-15083), otherwise, we have to update all the connectors.